### DISCUSSION TITLE
build error for ROS2 on windows 10, VS2019

### DISCUSSION THREAD

**Discussion by mkgeo** [gabime/spdlog#2236] [Q&A]
I got below errors durning cmake build, when I look at the circular_q.h I don't see any special character in that. I got the source files stright from the git. Any guys have ideas about this problem? 

C:\dev\ros2_foxy\build\spdlog_vendor\spdlog-1.5.0-prefix\src\spdlog-1.5.0\include\spdlog/details/circular_q.h(1,1): error C2220: the following warning is treated as an error (compiling source file C:\dev\ros2_foxy\build\spdlog_vendor\spdlog-1.5.0-prefix\src\spdlog-1.5.0\src\color_sinks.cpp) [C:\dev\ros2_foxy\build\spdlog_vendor\spdlog-1.5.0-prefix\src\spdlog-1.5.0-build\spdlog.vcxproj] [C:\dev\ros2_foxy\build\spdlog_vendor\spdlog-1.5.0.vcxproj]

C:\dev\ros2_foxy\build\spdlog_vendor\spdlog-1.5.0-prefix\src\spdlog-1.5.0\include\spdlog/details/circular_q.h(1,1): warning C4819: The file contains a character that cannot be represented in the current code page (950). Save the file in Unicode format to prevent data loss (compiling source file C:\dev\ros2_foxy\build\spdlog_vendor\spdlog-1.5.0-prefix\src\spdlog-1.5.0\src\async.cpp) [C:\dev\ros2_foxy\build\spdlog_vendor\spdlog-1.5.0-prefix\src\spdlog-1.5.0-build\spdlog.vcxproj] [C:\dev\ros2_foxy\build\spdlog_vendor\spdlog-1.5.0.vcxproj]

**Comment by tt4g** [comment]
Same problem answer: https://github.com/gabime/spdlog/issues/2012#issuecomment-886151094

Visual Studio determines the source code encoding from the OS language settings.
Visual Studio loads the source code as Code Page 950 (Chinese).

> C:\dev\ros2_foxy\build\spdlog_vendor\spdlog-1.5.0-prefix\src\spdlog-1.5.0\include\spdlog/details/circular_q.h(1,1): warning C4819: The file contains a character that cannot be represented in the current code page (950). Save the file in Unicode format to prevent data loss

Set compiler option `/source-charset:utf-8` (or `/utf8`) to change the character encoding of source files recognized by Visual Studio.
